### ISSUE TITLE
Slow response on login, getBalance, getSymbol

### ISSUE THREAD

**Issue by akbaralifreedom123** [Reiryoku-Technologies/Mida#139]
Hi, 
I want to create a copy trading scenario to place an order using same symbol for 1000s to 100,000s Ctrader accounts on the same time (possibly within a second).
What I am doing is, I am generating traffic towards my MIDA API using a .NET program that passes parameters to all the requests, and it is generating1000s of requests within milliseconds, and my ExpressJS endpoint is handling it gracefully without any delay. 

But problem arises when it tries to connect to Ctrader to fetch balance, symbol pip value etc it is taking very long (roughly a minute per 100s request on average) that makes my parallel execution so slow that it becomes unacceptable. Please provide insight into it, here is my sample code for more insights.

router.post('/createorder', async (req: express.Request, res: express.Response) => {
    try {
        const clientId = req.body.clientId;
        const clientSecret = req.body.clientSecret;
        const accessToken = req.body.accessToken;
        const accountId = req.body.accountId;
        const symbol = req.body.symbol;
        const direction = req.body.direction;

        // 1) Login, or fetch from Cache
        const myAccount = await login("cTrader", {
            clientId: clientId,
            clientSecret: clientSecret,
            accessToken: accessToken,
            accountId: accountId,
        });

        // 2) Get latest Balance
        const balance = await myAccount.getBalance();

        //3) Get latest symbol
        const liveSymbol = await myAccount.getSymbol(symbol);
        const liveSymbolAvg = await liveSymbol?.getAverage();

        //4) Perform clculations (SL/TP and volume) based on latest balance, symbol pip values etc
        const volume = decimal;
        const takeProfit = decimal;
        const stopLoss = decimal;

        //5) Finally place order
        const myOrder = await myAccount.placeOrder({
            symbol: symbol,
            direction: direction,
            volume: volume,
            protection: {
                takeProfit: takeProfit,
                stopLoss: stopLoss,
            },
        });
        res.jsonp(myOrder);
     
    } catch (err) {
        console.log(err);
        res.status(500).jsonp(err);
    }
});

**Comment by orange1337** [comment]
@akbaralifreedom123 
Did you find any solution, interesting the same question ...

**Comment by akbaralifreedom123** [comment]
> 

no answer yet

**Comment by orange1337** [comment]
> > 
> 
> no solution/answer found yet, but we are making mix and match both with OpenAPI (.Net) and MIDA

Seems like this problem is related to the cTrader side, btw do you use your own cTrader server, is there any rate limits on the cTrader side, or is something missed with the setup? 
I didn't integrate with cTrade before and didn't set it up ...

**Comment by akbaralifreedom123** [comment]
> 
I agree this seems an issues with the ctrader if you throw a lot of traffic especially login->getbalance etc, if you have already logged in accounts, then rest of the functions take less time. Ctrader does have some rate limiting in place but that is per user level, and I suspect they apply IP level too limiting too (I dont know and probably cannot verify it)

**Comment by akbaralifreedom123** [comment]
but I can add a point here, if I do the same stuff using ctrader OpenAPI (.net) I can get much better performance comparitively.

**Comment by Vasile-Peste** [comment]
@akbaralifreedom123 are you on the Discord or Telegram community? It's possible to discuss a little bit in private?

Meanwhile, we acknowledge this issue and there are many sides where we can improve, I think we can make it even faster than .NET! Thank you!

**Comment by desoss** [comment]
Hi guys, do you have any updates regarding the performance improvements? Or are there any plans to improve them in the future?
I'm highly interested in this topic.
